### PR TITLE
Rollup config to ts

### PR DESCRIPTION
**Pull request by birkskyum**
I want to take smaller steps, so this PR just moves the build files, incl. rollup config, to typescript, but doesn't rework the whole build system.

This corresponds to the two first commits of #961 , before everything went sour due to the web-worker.

### CODE REVIEW

**Comment by birkskyum** [comment]
I want to take smaller steps, because the web worker is quite a headache. This moves the files to typescript - then we can fix the web worker afterwards.

File: package.json [hunk 1]
```diff
@@ -149,18 +149,17 @@
     "build-swc-bench": "swc bench -d rollup/build/tsc/bench -D -s",
     "build-tsc": "npm run build-swc-src && npm run build-swc-bench",
     "build-tsc-old": "tsc --outDir rollup/build/tsc",
-    "build-dev": "npm run build-tsc && rollup -c --environment BUILD:dev",
-    "watch-dev": "rollup -c --environment BUILD:dev --watch",
-    "build-prod": "npm run build-tsc && rollup -c --environment BUILD:production",
-    "build-prod-min": "npm run build-tsc && rollup -c --environment BUILD:production,MINIFY:true",
-    "build-csp": "rollup -c rollup.config.csp.js",
-    "build-query-suite": "rollup -c test/integration/rollup.config.test.js",
+    "build-dev": "npm run build-tsc && rollup --configPlugin @rollup/plugin-typescript -c --environment BUILD:dev",
```
Comment (HarelM) [inline package.json:152]: Is the plugin here needed in order to use configuration file that are written in ts?
I don't see a lot of value for this change if this is the case.
For me it's OK that the config files are in javascript...

Comment (birkskyum) [reply package.json:152]: Yes, the plugin is needed for that, and any files imported from the config as well.

Alright. We only lint ts files, and the language server can help us surface small inconsistencies made by accident. I think that's the only advantage.

Comment (HarelM) [reply package.json:152]: Fair enough. Most examples are in JS and I hope it won't prove problematic. In any case, let's continue forward. We can always revert that part if needed.

**Review by HarelM — APPROVED** [review]

**Pull request merged**